scheduler/tasks.py

```diff
@@ -88,19 +91,19 @@ def run(self, schedule_type, lookup_id, **kwargs):
         l.info("Queuing <%s> <%s>" % (schedule_type, lookup_id))
 
         task_run = QueueTaskRun()
-        task_run.task_id = self.request.id
+        task_run.task_id = self.request.id or uuid4()
         task_run.started_at = now()
         tr_qs = QueueTaskRun.objects
 
         # Load the schedule active items
-        schedules = Schedule.objects.filter(enabled=True)
+        query = "SELECT id FROM scheduler_schedule WHERE enabled = 't'"
         if schedule_type == "crontab":
-            schedules = schedules.filter(celery_cron_definition=lookup_id)
+            query = query + 'AND celery_cron_definition_id = %(lookup_id)s'
             tr_qs = tr_qs.filter(celery_cron_definition=lookup_id)
             scheduler_type = CrontabSchedule
             task_run.celery_cron_definition_id = lookup_id
         elif schedule_type == "interval":
-            schedules = schedules.filter(celery_interval_definition=lookup_id)
+            query = query + 'AND celery_interval_definition_id = %(lookup_id)s'
```

Comment:
Not hugely important, but what about using `.query` on the queryset, rather than writing the SQL ourselves?